lib/Dialect/RTL/RTLDialect.cpp

```diff
@@ -16,16 +16,69 @@
 #include "mlir/IR/Builders.h"
 #include "mlir/IR/BuiltinTypes.h"
 #include "mlir/IR/DialectImplementation.h"
+#include "mlir/Transforms/InliningUtils.h"
 
 using namespace circt;
 using namespace rtl;
 
 //===----------------------------------------------------------------------===//
-// Dialect specification.
+// Inliner Dialect Interface
 //===----------------------------------------------------------------------===//
 
 namespace {
+struct RTLInlinerInterface : public mlir::DialectInlinerInterface {
+  using mlir::DialectInlinerInterface::DialectInlinerInterface;
+
+  /// Returns true is the `callable` operation can be inlined into the location
+  /// of the `call` operation.
+  bool isLegalToInline(Operation *call, Operation *callable,
+                       bool wouldBeCloned) const override final {
+    // Only inline an  operation which is marked inline
+    return call->hasAttr("inline");
+  }
 
+  /// Returns true if the operation from RTL `op` can be inlined in to the
+  /// destination region `dest.
+  bool
+  isLegalToInline(Operation *op, Region *dest, bool,
+                  BlockAndValueMapping &valueMapping) const override final {
+    // No operations in RTL prevent inlining.
+    return true;
+  }
```

Comment:
Is op always an rtl or comb op and dest always an rtl region?  might you see dest be a func or something?  How does this reject an sv op in a rtl.module?

Reply:
> Is op always an rtl or comb op and dest always an rtl region? 

The `op` will always be an RTL operation being inlined from a mysterious source region (probably an `rtl.module` body) in to some mysterious `dest` region from any dialect (probably from the callsite of an `rtl.instance`).

The Comb dialect has to separately implement this interface to allow Comb operations to be inlined.

> How does this reject an sv op in a rtl.module?

This particular bit of code only handles RTL ops. Since the SV dialect does not implement the inliner interface, no SV ops can be inlined.